sympy/codegen/rewriting.py

```diff
@@ -181,9 +181,10 @@ def __init__(self, func, func_m_1):
         self.func_m_1 = func_m_1
 
     def _try_func_m_1(self, expr):
-        protected, old_new =  expr.replace(self.func, lambda arg: Dummy(), map=True)
+        old_new = {}
+        protected = expr.replace(self.func, lambda arg: old_new.setdefault(arg, Dummy()))
```

Comment:
Beautiful application of this dict method!

Reply:
Thanks! :smiley: